### PR TITLE
Feat: Implement API Endpoint for Adding Blog Categories

### PR DESCRIPTION
**Pull request by CallmeInnocent**
### How to Test
Access the `POST /api/v1/create/blog-categories` endpoint with category name
Expected result: The category is created successfully.
Example response: 
```java
{
  "status": "success",
  "message": "Blog category created successfully.",
  "data": {
    "name": "Category Name"
  }
}

``` 

Access the `POST /api/v1/create/blog-categories` endpoint without authentication.
Expected result: A 401 error is returned, indicating invalid authentication credentials.
Example response:

```java
 {
    "status_code": 401
    "message": "The JWT signature is invalid",
    "error": "JWT signature does not match locally computed signature. JWT validity cannot be asserted and should not be trusted.",
}

``` 

Access the `POST /api/v1/ api/v1/create/blog-categories` endpoint when not SUPERADMIN 
Expected result: A 403 error is returned, indicating unauthorized access.
```java
Example response: { "statusCode": 403, 
"message": "You do not have the necessary permissions to access this resource" }
```

Access the` POST /api/ api/v1/create/blog-categories ` endpoint and encounter a server error.
Expected result: A 500 error is returned.
```java
Example response: {
  "status": "error",
  "message": "Internal Server Error. Please try again later.",
  "status_code": 500
}
``` 

Access the `POST /api/v1/create/blog-categories` endpoint with EMPTY FIELD.
Expected result: Validation Error.

```java
Example response: {
"status_code": 422,
    "error": "validation error",
    "detail": {
        "name": "Name is required"
    }
}
``` 

Reference Issues
[FEAT]: Implement Endpoint to Retrieve About Page Content (#265)
https://github.com/hngprojects/hng_boilerplate_nestjs/issues/142

### CODE REVIEW

File: src/main/java/hng_java_boilerplate/user/enums/Role.java [hunk 1]
```diff
@@ -1,5 +1,5 @@
 package hng_java_boilerplate.user.enums;
 
 public enum Role {
-    ROLE_USER, ROLE_ADMIN
+    ROLE_USER, ROLE_ADMIN, ROLE_SUPER_ADMIN
```
Comment (lurldgbodex) [inline src/main/java/hng_java_boilerplate/user/enums/Role.java:4]: what's the difference between admin and super_admin?

File: src/main/resources/application.properties [hunk 2]
```diff
@@ -1,25 +1,28 @@
 spring.application.name=hng-java-boilerplate
```
Comment (lurldgbodex) [inline src/main/resources/application.properties:2]: delete the applicaiton.properties file

File: src/main/java/hng_java_boilerplate/user/enums/Role.java (hunk 1 above)
Comment (CallmeInnocent) [reply src/main/java/hng_java_boilerplate/user/enums/Role.java:4]: Super admin can assign admin roles to other users.

Comment (CallmeInnocent) [reply src/main/java/hng_java_boilerplate/user/enums/Role.java:4]: Can do things other admins can't do.

File: src/main/resources/application.properties (hunk 2 above)
Comment (CallmeInnocent) [reply src/main/resources/application.properties:2]: Okay

File: src/main/resources/db/migration/V12__create_category_table.sql [hunk 3]
```diff
@@ -0,0 +1,7 @@
+-- V12__Create_category_table.sql
```
Comment (lurldgbodex) [inline src/main/resources/db/migration/V12__create_category_table.sql:1]: you need to update the migration version

**Review by lurldgbodex — CHANGES_REQUESTED** [review]

File: src/main/java/hng_java_boilerplate/blogCategory/entity/Category.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    @Id
+    private String id;
+
+    private String name;
+
+//    private String testCategory;
+
+//    public Category(String id, String name) {
+//        this.id = id;
+//        this.name = name;
```
Comment (lurldgbodex) [inline src/main/java/hng_java_boilerplate/blogCategory/entity/Category.java:26]: delete commented out code

File: src/main/java/hng_java_boilerplate/blogCategory/service/CreateBlogCategoryService.java [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class CreateBlogCategoryService {
+
+    @Autowired
+    CreateBlogCategoryRepository categoryRepository;
```
Comment (lurldgbodex) [inline src/main/java/hng_java_boilerplate/blogCategory/service/CreateBlogCategoryService.java:23]: constructor injection?

**Pull request closed without merging**